File: 0032-longest-valid-parentheses/0032-longest-valid-parentheses.py
class Solution:
    def longestValidParentheses(self, s: str) -> int:
        maxLength = 0
        st = [-1] # Base for calculating the length
        
        for i in range(len(s)):
            if s[i] == '(':
                st.append(i)
            else:
                st.pop()
                if not st:
                    st.append(i) # Update the base for length calculation
                else:
                    maxLength = max(maxLength, i - st[-1])
        
        return maxLength

File: 0032-longest-valid-parentheses/0032-longest-valid-parentheses.swift
class Solution {
    func longestValidParentheses(_ s: String) -> Int {
        var maxLength = 0
        var stack = [-1] // Base for calculating the length
        
        for (i, char) in s.enumerated() {
            if char == "(" {
                stack.append(i)
            } else {
                stack.removeLast()
                if stack.isEmpty {
                    stack.append(i) // Update the base for length calculation
                } else {
                    maxLength = max(maxLength, i - stack.last!)
                }
            }
        }
        return maxLength
    }
}